src/dune_rules/workspace.ml

```diff
@@ -324,6 +324,22 @@ module Context = struct
       ; dynamically_linked_foreign_archives = true
       ; instrument_with = Option.value instrument_with ~default:[]
       }
+
+  let build_contexts t =
+    let name = name t in
+    let native =
+      Build_context.create ~name ~host:(host_context t)
+        ~build_dir:(Path.Build.of_string (Context_name.to_string name))
```

Comment:
There's `Context_name.build_dir` to simplify this.

Reply:
Actually, do we need to pass `build_dir` at all?  `create` can figure it out from the name.

Reply:
indeed, changing that